python/src/array.cpp

```diff
@@ -39,6 +39,10 @@ py::list to_list(array& a, size_t index, int dim) {
 }
 
 auto to_scalar(array& a) {
+  {
+    py::gil_scoped_release nogil;
```

Comment:
Does it make sense to add these to the `save*` functions as well? They all do evals.